core/src/main/java/io/questdb/cairo/ColumnVersionWriter.java

```diff
@@ -88,20 +89,58 @@ public boolean hasChanges() {
     public void removeColumnTop(long partitionTimestamp, int columnIndex) {
         int recordIndex = getRecordIndex(partitionTimestamp, columnIndex);
         if (recordIndex >= 0) {
-            getCachedList().setQuick(recordIndex + 3, 0);
+            cachedList.setQuick(recordIndex + COLUMN_TOP_OFFSET, 0);
+            hasChanges = true;
+        }
+    }
+
+    public void detachPartitionColumns(long partitionTimestamp) {
```

Comment:
this method removed entire partition, perhaps it should be called "removePartition()" ?

Reply:
done in [cfb28c6](https://github.com/questdb/questdb/pull/2280/commits/cfb28c677445ec4750dd249b53f50337dce48f48)